src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -21,569 +24,274 @@
-            These can be, for example, data assets that were registered.
-
-        Returns
-        -------
-        requests.Response
-
-        """
-        data_assets = []
-        if run_capsule_config.data_assets is not None:
-            # check if data assets exist
-            data_assets += run_capsule_config.data_assets
-        if input_data_assets is not None:
-            data_assets += input_data_assets
-        self.check_data_assets(data_assets)
-
-        run_capsule_request = RunCapsuleRequest(
-            capsule_id=run_capsule_config.capsule_id,
-            pipeline_id=run_capsule_config.pipeline_id,
-            data_assets=data_assets,
-            parameters=run_capsule_config.run_parameters,
-            version=run_capsule_config.capsule_version,
+        run_capsule_response = self.api_handler.co_client.run_capsule(
+            self.process_config
         )
-        # TODO: Handle case of bad response from code ocean
-        run_capsule_response = self.co_client.run_capsule(run_capsule_request)
         run_capsule_response_json = run_capsule_response.json()
+
+        if run_capsule_response_json.get("id") is None:
+            raise KeyError(
+                f"Something went wrong running the capsule or pipeline. "
+                f"Response Status Code: {run_capsule_response.status_code}. "
+                f"Response Message: {run_capsule_response_json}"
+            )
+
         computation_id = run_capsule_response_json["id"]
 
         # TODO: We may need to clean up the loop termination logic
-        if run_capsule_config.pause_interval:
+        if self.process_poll_interval_seconds:
             executing = True
             num_checks = 0
             while executing:
                 num_checks += 1
-                time.sleep(run_capsule_config.pause_interval)
-                computation_response = self.co_client.get_computation(
-                    computation_id
+                time.sleep(self.process_poll_interval_seconds)
+                computation_response = (
+                    self.api_handler.co_client.get_computation(computation_id)
                 )
                 curr_computation_state = computation_response.json()
 
                 if (curr_computation_state["state"] == "completed") or (
-                    (run_capsule_config.timeout_seconds is not None)
+                    (self.process_timeout_seconds is not None)
                     and (
-                        run_capsule_config.pause_interval * num_checks
-                        >= run_capsule_config.timeout_seconds
+                        self.process_poll_interval_seconds
+                        * num_checks
+                        >= self.process_timeout_seconds
                     )
                 ):
                     executing = False
         return run_capsule_response
 
-    def _register_data_and_update_permissions(
-        self, register_data_config: RegisterDataConfig
+    def capture_result(
```

Comment:
@dyf I don't see where input tags and custom metadata are propagated to the processed asset. Has this feature been removed?